final_project.py

```diff
@@ -0,0 +1,369 @@
+Mercedes is connected to Walter, Robin, Bryant.\
+Mercedes likes to play The Legend of Corgi, Pirates in Java Island, Seahorse Adventures.\
+Olive is connected to John, Ollie.\
+Olive likes to play The Legend of Corgi, Starfleet Commander.\
+Debra is connected to Walter, Levi, Jennie, Robin.\
+Debra likes to play Seven Schemers, Pirates in Java Island, Dwarves and Swords.\
+Walter is connected to John, Levi, Bryant.\
+Walter likes to play Seahorse Adventures, Ninja Hamsters, Super Mushroom Man.\
+Levi is connected to Ollie, John, Walter.\
+Levi likes to play The Legend of Corgi, Seven Schemers, City Comptroller: The Fiscal Dilemma.\
+Ollie is connected to Mercedes, Freda, Bryant.\
+Ollie likes to play Call of Arms, Dwarves and Swords, The Movie: The Game.\
+Jennie is connected to Levi, John, Freda, Robin.\
+Jennie likes to play Super Mushroom Man, Dinosaur Diner, Call of Arms.\
+Robin is connected to Ollie.\
+Robin likes to play Call of Arms, Dwarves and Swords.\
+Freda is connected to Olive, John, Debra.\
+Freda likes to play Starfleet Commander, Ninja Hamsters, Seahorse Adventures."
+
+# ----------------------------------------------------------------------------- 
+# create_data_structure(string_input): 
+#   Parses a block of text (such as the one above) and stores relevant 
+#   information into a data structure. You are free to choose and design any 
+#   data structure you would like to use to manage the information.
+# 
+# Arguments: 
+#   string_input: block of text containing the network information
+#
+#   You may assume that for all the test cases we will use, you will be given the 
+#   connections and games liked for all users listed on the right-hand side of an
+#   'is connected to' statement. For example, we will not use the string 
+#   "A is connected to B.A likes to play X, Y, Z.C is connected to A.C likes to play X."
+#   as a test case for create_data_structure because the string does not 
+#   list B's connections or liked games.
+#   
+#   The procedure should be able to handle an empty string (the string '') as input, in
+#   which case it should return a network with no users.
+# 
+# Return:
+#   The newly created network data structure
+def create_data_structure(s):
+    network = {} # {user1:[[connections],[games]], user2:[[connections],[games]]...} 
+    user = []
+    connection = []
+    game = []
+    string_ls = s.split('.')
+    #print string_ls
+    if len(s) > 1:
+        
+        for e in string_ls:
+        
+            if 'is connected to' in e:
+                s = e[e.find('is connected to') + 16:] #s is the names of the
+                #connection and the next string to be used
+                #print s
+                s = s.split(', ')
+                user_name = e[:e.find(' is connected to')]
+                user.append(user_name)
+                connection.append(s)
+                #print connection, user
+            
+            if 'likes to play' in e:
```

Comment:
hm.. so this works because they are giving you sentences in order. but a more robust way of doing this would be to read the user from both sentences. in this case you're ignoring the user in the "likes to play" sentence since you're assuming its the same as in the sentence before that, right? in general doesn't seem like a good assumption to make and its not listed in the above assumptions.

a more robust approach would be the following:

```
for statement in statements:
    if 'is connected to' in e:
        # parse the user and friends
       connections[user] = friends

    if 'likes to play' in e:
        # parse user & games
        games[user] = <parsed games>

````

high level , the main difference is that you're adding to the dictionary right away, as opposed to putting in a list first and then doing a second pass to create the dict. there is no reason you can't build up the dictionary in the first pass itself.